src/edu/psu/sweng500/team8/gui/PencilMarkGridPanel.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package edu.psu.sweng500.team8.gui;
+
+import java.awt.Color;
+import java.awt.Component;
+import java.awt.Dimension;
+
+import javax.swing.JPanel;
+import javax.swing.JTextField;
+import javax.swing.JToggleButton;
+import javax.swing.text.BadLocationException;
+import javax.swing.text.DefaultHighlighter;
+import javax.swing.text.Highlighter.HighlightPainter;
+
+import edu.psu.sweng500.team8.coreDataStructures.Cell;
+import edu.psu.sweng500.team8.coreDataStructures.Cell.ValueType;
+import edu.psu.sweng500.team8.play.GameSession;
+
+public class PencilMarkGridPanel extends javax.swing.JPanel {
+	// private JPanel board;
+	private JPanel[][] pencilMarkCell = new JPanel[9][9];
+	private GameSession gameSession;
+
+	public PencilMarkGridPanel() {
+
+	}
+
+	public void populate(GameSession gameSession) {
+		this.gameSession = gameSession;
+		initComponents();
+	}
+
+	private void initComponents() {
+
+		this.removeAll();
+
+		this.setPreferredSize(new java.awt.Dimension(500, 500));
+		this.setLayout(new java.awt.GridLayout(9, 9));
+
+		for (int i = 0; i < 9; i++) {
+			for (int j = 0; j < 9; j++) {
+				Cell cell = this.gameSession.getGameBoard().getCellGrid()
+						.getCell(i, j);
+				if (cell.hasNumber()) {
+
+					JTextField textField = new JTextField();
+
+					textField
+							.setHorizontalAlignment(javax.swing.JTextField.CENTER);
+					textField.setBorder(javax.swing.BorderFactory
+							.createLineBorder(new java.awt.Color(0, 0, 0)));
+					textField.setText(Integer.toString(cell.getNumber()));
+
+					HighlightPainter painter = new DefaultHighlighter.DefaultHighlightPainter(
+							Color.green);
+
+					try {
+						textField.getHighlighter().addHighlight(0, 3, painter);
+					} catch (BadLocationException e) {
+						e.printStackTrace();
+					}
+
+					textField.setEditable(false);
+
+					textField.setPreferredSize(new Dimension(18 * 3, 18 * 3));
+
+					pencilMarkCell[i][j] = new JPanel();
+
+					pencilMarkCell[i][j].setPreferredSize(new Dimension(18 * 3,
+							18 * 3));
+					pencilMarkCell[i][j].setBorder(null);
+
+					pencilMarkCell[i][j].add(textField);
+					this.add(pencilMarkCell[i][j]);
+
+				} else {
```

Comment:
There's a bit of duplicated code in both cases. Suggest cleaning this up a bit. Use helper methods as appropriate.